### DISCUSSION TITLE
using madsim 0.2.22，dlsym load an empty pointer

### DISCUSSION THREAD

**Discussion by lyc200150** [madsim-rs/madsim#154] [Q&A] [answered]
hi, I am a beginner for madsim. I can use madsim at version 0.2.0-alpha.4，but when I use madsim at version 0.2.22 to test my project,  there will throw an error: thread 'main' panicked at 'assertion failed: !ptr.is_null()', /root/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/madsim-0.2.22/src/sim/rand.rs:221:17，which is let ptr = libc::dlsym(libc::RTLD_NEXT, b"getrandom\0".as_ptr() as _); load an empty pointer. how can I fix this problem, thanks for your help

**Comment by wangrunji0408** [comment, marked as answer]
Hi, thanks for your feedback. Madsim intercepts the libc `getrandom` function to achieve determinism in simulations. This line is used for calling the original function when we are not in simulation. It seems that madsim couldn't locate the original function in your environment. Could you provide more context about your environment? e.g. OS, libc version, static or dynamic link... That would be helpful for troubleshooting. Thanks!

**Reply by lyc200150** [reply]
I am developing in a docker, which has centos 7.9.2009，ldd version is 2.17. the libc.so.6 is at /usr/lib64，I have add it to the path，but when I try to see the symbols of the libc.so，I find there isn't exist a symbol "getrandom"，maybe I should update my libc.so?

**Reply by wangrunji0408** [reply]
Sure. Perhaps you can install the debuginfo package for libc and try again.

**Reply by lyc200150** [reply]
I update my libc.so to 2.27 and it works, thanks for your help!
